APCAppCore/APCAppCore/UI/Onboarding/SignUp/APCSignUpGeneralInfoViewController.m

```diff
@@ -617,6 +617,61 @@ - (void) validateContent {
     }
 }
 
+- (void) signUp {
+    APCSpinnerViewController *spinnerController = [[APCSpinnerViewController alloc] init];
+    [self presentViewController:spinnerController animated:YES completion:nil];
+    
+    typeof(self) __weak weakSelf = self;
+    [self.user signUpOnCompletion:^(NSError *error) {
+        if (error) {
+            
+            APCLogError2 (error);
+            
+            if (error.code == SBBErrorCodeInternetNotConnected || error.code == SBBErrorCodeServerNotReachable || error.code == SBBErrorCodeServerUnderMaintenance) {
+                [spinnerController dismissViewControllerAnimated:NO completion:^{
+                    
+                    UIAlertController *alertView = [UIAlertController alertControllerWithTitle:NSLocalizedStringWithDefaultValue(@"Sign Up", @"APCAppCore", APCBundle(), @"Sign Up", @"")
+                                                                                       message:error.localizedDescription
+                                                                                preferredStyle:UIAlertControllerStyleAlert];
+                    
+                    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+                                                                          handler:^(UIAlertAction * __unused action) {}];
+                    
+                    [alertView addAction:defaultAction];
+                    [self presentViewController:alertView animated:YES completion:nil];
```

Comment:
`weakSelf`

In general, I find that for large completion blocks, it's safer to call out to a method `[weakSelf handleSignUp:error]`